spark/firmware
# Feature/vendorlibraries

Adds an `APPLIBS` variable that defines the location of 'tweaked' particle libraries.

To test, 

- download the [freertso4core](https://github.com/m-mcgowan/particle-freertos4core) library 
- create a new directory /usr/local/myapp for an application, separate from the library directory, and copy the` timer.cpp` example from the library to that directory.
- perform the library tweaks as described in the build documentation (in this PR)
- build the example file with the app by running
```
make APPDIR=/usr/local/myapp APPLIBS=/path/to/freertos4core
```

The build should succeed. 

---

Doneness:

- [x] Contributor has signed CLA
- [ ] API tests compiled
- [ ] Add to CHANGELOG.md after merging (add links to docs and issues)
